### PR TITLE
Using keywords for ivadomed/scripts folder

### PR DESCRIPTION
**Pull request by cakester**
#### GitHub

- [x] I've given this PR a concise, self-descriptive, and meaningful title
- [x] I've linked relevant issues in the PR body
- [x] I've applied [the relevant labels](https://www.neuro.polymtl.ca/software/contributing#pr_labels) to this PR
- [x] I've assigned a reviewer

#### PR contents

- [x] I've consulted [ivadomed's internal developer documentation](https://github.com/ivadomed/ivadomed/wiki) to ensure my contribution is in line with any relevant design decisions
- [x] I've added [relevant tests](https://github.com/ivadomed/ivadomed/wiki/tests) for my contribution
- [x] I've updated the [relevant documentation](https://github.com/ivadomed/ivadomed/wiki/documentation) for my changes, including argparse descriptions, docstrings, and ReadTheDocs tutorial pages

## Description

using keywords for ivadomed/scripts folder
This is the continuation of PR #904 focusing primarily on script module (being updated).

### CODE REVIEW

File: ivadomed/scripts/visualize_transforms.py [hunk 1]
```diff
@@ -152,9 +154,9 @@ def run_visualization(input, config, number, output, roi):
                                               data_type="im")
 
             # Plot before / after transformation
-            fname_out = str(Path(output, stg_transforms+"slice"+str(i)+".png"))
-            print("Fname out: {}.".format(fname_out))
-            print("\t{}".format(dict(metadata)))
+            fname_out = str(Path(output, stg_transforms + "slice" + str(i) + ".png"))
+            print(f"Fname out: {fname_out}.")
+            print(f"\t{dict(metadata)}")
```
Comment (dyt811) [inline ivadomed/scripts/visualize_transforms.py:159]: Could you try to see which `logger info/debug` we can replace the `print` with here? I thought we replaced all print statements.

**Review by dyt811 — CHANGES_REQUESTED** [review]
Please ensure the import statement correction is made. Thanks a lot for the great work.

File: ivadomed/object_detection/utils.py [hunk 2]
```diff
@@ -237,22 +238,23 @@ def load_bounding_boxes(object_detection_params, subject_path_list, slice_axis,
     """
     # Load or generate bounding boxes and save them in json file
     bounding_box_dict = {}
-    if object_detection_params is None or object_detection_params['object_detection_path'] is None:
+    if object_detection_params is None or object_detection_params[ObjectDetectionParamsKW.OBJECT_DETECTION_PATH] is None:
         return bounding_box_dict
-    bounding_box_path = Path(object_detection_params.get('path_output'), 'bounding_boxes.json')
+
+    bounding_box_path = Path(object_detection_params.get(ObjectDetectionParamsKW.PATH_OUTPUT), 'bounding_boxes.json')
```
[the remark this answers is not in the document]
Comment (cakester) [reply ivadomed/object_detection/utils.py:244]: $ grep bounding_boxes.json .
grep: warning: GREP_OPTIONS is deprecated; please use an alias or script
./ivadomed/object_detection/utils.py:128:    file_path = Path(path_output, 'bounding_boxes.json')
./ivadomed/object_detection/utils.py:244:    bounding_box_path = Path(object_detection_params.get(ObjectDetectionParamsKW.PATH_OUTPUT), 'bounding_boxes.json')
./testing/unit_tests/test_bounding_box.py:82:    bounding_box_path = os.path.join(PATH_OUTPUT, 'bounding_boxes.json')

just used 3 times

File: ivadomed/scripts/automate_training.py [hunk 3]
```diff
@@ -154,33 +154,33 @@ def split_dataset(initial_config):
...
         path_output="./",
-        balance=initial_config["split_dataset"]['balance'] \
-        if 'balance' in initial_config["split_dataset"] else None
+        balance=initial_config[ConfigKW.SPLIT_DATASET][SplitDatasetKW.BALANCE] \
+        if SplitDatasetKW.BALANCE in initial_config[ConfigKW.SPLIT_DATASET] else None
     )
 
     # save the subject distribution
     split_dct = {'train': train_lst, 'valid': valid_lst, 'test': test_lst}
     split_path = "./" + "common_split_datasets.joblib"
```
[the remark this answers is not in the document]
Comment (cakester) [reply ivadomed/scripts/automate_training.py:181]: $ grep \'train\' .
grep: warning: GREP_OPTIONS is deprecated; please use an alias or script
./dev/class_balance.py:57:    for ds_lst, ds_name in zip([train_lst, valid_lst, test_lst], ['train', 'valid', 'test']):
./dev/metadata_config.py:24:    for subset in ['train', 'validation', 'test']:
./dev/plot_cluster_metadata.py:39:    for s, y_val in zip(['train', 'valid', 'test'], [0.25, 0.5, 0.75]):
./dev/plot_cluster_metadata.py:63:    for subset in ['train', 'valid', 'test']:
./dev/plot_cluster_metadata.py:85:        values = [v for s in ['train', 'valid', 'test'] for v in metadata_dct[s][m]]
./ivadomed/loader/utils.py:178:    split_dct = {'train': train_lst, 'valid': valid_lst, 'test': test_lst}
./ivadomed/loader/utils.py:200:        train_lst, valid_lst, test_lst = old_split['train'], old_split['valid'], old_split['test']
./ivadomed/main.py:36:    command_group.add_argument("--train", dest='train', action='store_true',
./ivadomed/main.py:385:    if command == 'train':
./ivadomed/main.py:423:        if command != 'train':  # If command == train, then ds_valid already load
./ivadomed/main.py:450:    if command == 'train':
./ivadomed/scripts/automate_training.py:180:    split_dct = {'train': train_lst, 'valid': valid_lst, 'test': test_lst}

**Review by dyt811 — APPROVED** [review]
All good thanks a lot. We should keep an eye out for future simple text "train" "test" as they pop up in various contexts and would benefit more from context specific keywords conversion. We can leave them as it is for now. Might need a future PR/issues to specifically review that.

**Pull request merged**